packages/vm/lib/buildBlock.ts

```diff
@@ -0,0 +1,241 @@
+import { Address, BN, toBuffer } from 'ethereumjs-util'
+import { encode } from 'rlp'
+import { BaseTrie as Trie } from 'merkle-patricia-tree'
+import { TypedTransaction } from '@ethereumjs/tx'
+import { Block, BlockOptions, HeaderData } from '@ethereumjs/block'
+import VM from '.'
+import { RunTxResult } from './runTx'
+import Bloom from './bloom'
+import { generateTxReceipt, calculateMinerReward, rewardAccount } from './runBlock'
+
+/**
+ * Options for building a block.
+ */
+export interface BuildBlockOpts {
+  /**
+   * The parent block
+   */
+  parentBlock: Block
+
+  /**
+   * The block header data to use.
+   * Defaults used for any values not provided.
+   */
+  headerData?: HeaderData
+
+  /**
+   * The block options to use.
+   */
+  blockOpts?: BlockOptions
+}
+
+/**
+ * Options for sealing a block.
+ */
+export interface SealBlockOpts {
+  /**
+   * For PoW, the nonce.
+   * Overrides the value passed in the constructor.
+   */
+  nonce?: Buffer
+
+  /**
+   * For PoW, the mixHash.
+   * Overrides the value passed in the constructor.
+   */
+  mixHash?: Buffer
```

Comment:
Is it correct that we cannot seal blocks with clique using this new block builder API?

Reply:
We do support it.